src/acl/external/kerberos_sid_group/ext_kerberos_sid_group_acl.pl.in

```diff
@@ -0,0 +1,252 @@
+
+=head1 QUESTIONS
+
+Questions on the usage of this program can be sent to the I<Squid Users mailing list <    squid-users@lists.squid-cache.org>>
+
+=head1 REPORTING BUGS
+
+Bug reports need to be made in English.
+See http://wiki.squid-cache.org/SquidFaq/BugReporting for details of what you need to     include with your bug report.
+
+Report bugs or bug fixes using http://bugs.squid-cache.org/
+
+Report serious security bugs to I<Squid Bugs <squid-bugs@lists.squid-cache.org>>
+
+Report ideas for new improvements to the I<Squid Developers mailing list <squid-dev@li    sts.squid-cache.org>>
+
+=head1 SEE ALSO
+
+The Squid FAQ wiki http://wiki.squid-cache.org/SquidFaq
+
+The Squid Configuration Manual http://www.squid-cache.org/Doc/config/
+
+=cut
+
+#
+# Version history:
+#   2018-06-10 Markus Moeller <markus_moeller@compuserve.com>
+#               Initial release
+#
+# Globals
+#
+use File::Basename;
+use Date::Format;
+use vars qw/ %opt /;
+
+my $name = basename($0);
+my $principal;
+my $dc;
+my $basedn;
+my $ccname="/tmp/squid_krb5cc";
+my $groupSIDs;
+my @ADgroupSIDs;
+my $user;
+my @groups;
+my $ans;
+
+# Disable output buffering
+$|=1;
+
+sub debug()
+{
+	my @lt = localtime;
+	print STDERR strftime("%Y/%m/%d %H:%M:%S", @lt)." | $name: @_\n" if $opt{d};
+}
+
+sub info()
+{
+	my @lt = localtime;
+	print STDERR strftime("%Y/%m/%d %H:%M:%S", @lt)." | $name: @_\n"; 
+}
+
+sub check()
+{
+	if ( grep( /^@_$/, @ADgroupSIDs) ) {
+		&debug("DEBUG: Found @_ in AD group SID");
+		return "OK";
+	} else {
+		&debug("DEBUG: Did not find @_ in AD group SID");
+		return "ERR";
+	}   
+}
+
+#
+# Command line options processing
+#
+sub init()
+{
+	use Getopt::Std;
+	my $errmsg;
+	my $opt_string = 'hdD:p:b:G:';
+	getopts( "$opt_string", \%opt ) or usage();
+	usage() if $opt{h};
+	usage() if not defined $opt{D};
+	usage() if not defined $opt{b};
+	usage() if not defined $opt{p};
+	usage() if not defined $opt{G};
+
+	$ENV{'KRB5CCNAME'} = $ccname;
+
+	@groups = split(/:/,$opt{G});
+	$errmsg=`kinit -k $opt{p} 2>&1`;
+	&info("ERROR: $errmsg") if $errmsg;
+	exit 99 if $errmsg;
+
+	$errmsg="";
+	foreach my $group (@groups) {
+		open(LDAP, "ldapsearch -LLL -Ygssapi -H ldap://$opt{D}:389 -s sub -b \"$opt{b}\" \"(CN=$group)\" objectsid 2>&1 |");
+		my $sid;
+		while (<LDAP>) {
+			chomp($_);
+			if ( $_ =~ /^object/ && defined $sid ) {
+				&info("ERROR: multiple SIDs returned for group $group");
+			} elsif ( $_ =~ /^object/ ) {
+				$sid=$_;
+				$sid=~s/^[^\s]+\s+//;
+			} else {
+				$errmsg=$errmsg.";".$_;
+			}
+		}
+		close(LDAP);
+		if ( ! defined $sid ) {
+			$errmsg=~s/^;//;
+			&info("ERROR: $errmsg");
+			&info("ERROR: no SID returned for group $group");
+		} else {
+			&info("INFO:ldapsearch result Group=$group, SID=$sid");
+			push @ADgroupSIDs, $sid;
+		}
+	}
+	&info("ERROR: Exit as no sid was found for any group") if ! @ADgroupSIDs;
+	exit 99 if ! @ADgroupSIDs;
+}
+
+#
+# Message about this program and how to use it
+#
+sub usage()
```

Comment:
you are pulling in Pod::Usage which generates the help output from documentation SYNOPSIS section. No need to define a custom usage() function.